### ISSUE TITLE
`changes` object passed into Connection.modify() is mutated

### ISSUE THREAD

**Issue by jerrykan** [cannatag/ldap3#641]
When a `changes` object is passed to `Connection.modify()` it is mutated. Preferably any values passed into the method should remain unchanged.

```
import ldap3

server = ldap3.Server(...)
connection = ldap3.Connection(server, ...)
connection.open()

operations = {'givenName': ('MODIFY_REPLACE', ['Person'])}
connection.modify('cn=Person Two,ou=users,dc=testcase2,dc=org', operations)

# operations is now {'givenName': [('MODIFY_REPLACE', ['Person'])]}
```
I would have expected the `operations` dict to remain unchanged.

The mutation seems to occur at https://github.com/cannatag/ldap3/blob/a6bd3c78181b852f8111b5cff765f51871ed3ccb/ldap3/core/connection.py#L1057

**Comment by jerrykan** [comment]
I should note the current workaround I am using is:
```
from copy import deepcopy

...

connection.modify('cn=Person Two,ou=users,dc=testcase2,dc=org', deepcopy(operations))
```
but this seems like a bit of dodgy hack.

**Comment by cannatag** [comment]
You're right. Object passed to functions should not be mutated. Can you check the code in dev? It should remain unchanged.

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in version 2.6
